code/xcvm/cosmwasm/contracts/router/src/contract.rs

```diff
@@ -179,38 +278,75 @@ fn handle_instantiate_reply(deps: DepsMut, msg: Reply) -> StdResult<Response> {
 		)?)?
 	};
 
-	INTERPRETERS.save(deps.storage, (router_reply.0, router_reply.1), &interpreter_address)?;
+	match INTERPRETERS.load(deps.storage, (router_reply.0, router_reply.1.clone())) {
```

Comment:
It's clearer to deconstruct a tuple than using indexing like instead of `(x.0, x.1)` deconstruct at the function def with `(meaningful_name, meanigful_name_2): (A, B)`